packages/task/src/browser/task-configuration-manager.ts

```diff
@@ -142,50 +146,57 @@ export class TaskConfigurationManager {
         }
     }
 
-    protected async doOpen(model: TaskConfigurationModel): Promise<EditorWidget> {
+    protected async doOpen(model: TaskConfigurationModel): Promise<EditorWidget | undefined> {
         let uri = model.uri;
         if (!uri) {
             uri = await this.doCreate(model);
         }
-        return this.editorManager.open(uri, {
-            mode: 'activate'
-        });
+        if (uri) {
+            return this.editorManager.open(uri, {
+                mode: 'activate'
+            });
+        }
     }
 
-    protected async doCreate(model: TaskConfigurationModel): Promise<URI> {
-        await this.preferences.set('tasks', {}); // create dummy tasks.json in the correct place
-        const { configUri } = this.preferences.resolve('tasks'); // get uri to write content to it
-        let uri: URI;
-        if (configUri && configUri.path.base === 'tasks.json') {
-            uri = configUri;
-        } else { // fallback
-            uri = new URI(model.workspaceFolderUri).resolve(`${this.preferenceConfigurations.getPaths()[0]}/tasks.json`);
-        }
-        const content = this.getInitialConfigurationContent();
-        const fileStat = await this.filesystem.getFileStat(uri.toString());
-        if (!fileStat) {
-            throw new Error(`file not found: ${uri.toString()}`);
-        }
-        try {
-            await this.filesystem.setContent(fileStat, content);
-        } catch (e) {
-            if (!FileSystemError.FileExists.is(e)) {
-                throw e;
+    protected async doCreate(model: TaskConfigurationModel): Promise<URI | undefined> {
+        let uri = new URI(model.workspaceFolderUri).resolve(`${this.preferenceConfigurations.getPaths()[0]}/tasks.json`);
+        const existTheiaTaskConfig = await this.filesystem.exists(uri.toString());
```

Comment:
in order to check for an existence one must use `PreferenceService`, as far as I remember if a file does not exist it should be possible to resolve a uri

assuming some location is bogus, clients should not know it

Reply:
i just tried in the debugger. if a file does not exist, the uri is resolved as undefined.

if you look at the line 170 you would find I am not assuming uri because I used the preference service to resolve it at the end :)

Reply:
Why do you need `existTheiaTaskConfig`? Clients just should use uri resolved by the preference service.

Reply:
There is no even guruantee that `this.preferenceConfigurations.getPaths()[0]` is `.theia`. An end product can override it and return only `.vscode`. Please get rid of such assumptions in the client code.